include/CLI/Validators.hpp

```diff
@@ -240,6 +264,116 @@ struct Range : public Validator {
+    template <typename T, enable_if_t<is_copyable_ptr<T>::value, detail::enabler> = detail::dummy>
+    explicit IsMember(T set)
+        : IsMember(set,
+                   std::function<typename std::pointer_traits<T>::element_type::value_type(
+                       typename std::pointer_traits<T>::element_type::value_type)>{}) {}
+
+    /// This checks to see if an item is in a set: copy version. (Empty function)
+    template <typename T, enable_if_t<!is_copyable_ptr<T>::value, detail::enabler> = detail::dummy>
+    explicit IsMember(T set) : IsMember(set, std::function<typename T::value_type(typename T::value_type)>()) {}
+
+    /// This checks to see if an item is in a set: pointer version.
+    template <typename T, typename F, enable_if_t<is_copyable_ptr<T>::value, detail::enabler> = detail::dummy>
+    explicit IsMember(T set, F filter_function) {
+        using item_t = typename std::pointer_traits<T>::element_type::value_type;
+
+        std::function<item_t(item_t)> filter_fn = filter_function;
+
+        tname_function = [set]() {
+            std::stringstream out;
+            out << detail::type_name<item_t>() << " in {" << detail::join(*set, ",") << "}";
+            return out.str();
+        };
+
+        func = [set, filter_fn](std::string &input) {
+            auto result = std::find_if(std::begin(*set), std::end(*set), [filter_fn, input](item_t v) {
+                item_t a = v;
+                item_t b;
+                if(!detail::lexical_cast(input, b))
+                    throw ConversionError(input); // name is added later
+
+                if(filter_fn) {
+                    a = filter_fn(a);
+                    b = filter_fn(b);
+                }
+                return a == b;
+            });
+
+            if(result == std::end(*set)) {
+                return input + " not in {" + detail::join(*set, ",") + "}";
+            } else {
+                // Make sure the version in the input string is identical to the one in the set
+                // Requires std::stringstream << be supported on T.
+                std::stringstream out;
+                out << *result;
+                input = out.str();
+                return std::string();
+            }
+        };
+    }
+
+    /// This checks to see if an item is in a set: copy version.
+    template <typename T, typename F, enable_if_t<!is_copyable_ptr<T>::value, detail::enabler> = detail::dummy>
+    explicit IsMember(T set, F filter_function) {
+        using item_t = typename T::value_type;
+        std::function<item_t(item_t)> filter_fn = filter_function;
+
+        std::stringstream out;
+        out << detail::type_name<item_t>() << " in {" << detail::join(set, ",") << "}";
+        tname = out.str();
+
+        func = [set, filter_fn](std::string &input) {
+            auto result = std::find_if(std::begin(set), std::end(set), [filter_fn, input](item_t v) {
```

Comment:
This might look cleaner/simpler as a range based for loop instead of the find_if algorithm with a complex lambda, and you could collapse the else condition after result into the loop.  and a boolean found.    You also will only need to update the results if filter_fn actually contains something.

Reply:
or just have the return directly from the loop if found

Reply:
Done. I could split this into non-function versions too, so that `>>` would not be required on the type, but I think this is probably something that can be added later if anyone ever needs it.